### PR TITLE
Update instructions post oss-test-infra migration, as well as stop ge…

### PR DESCRIPTION
**Pull request by chaodaiG**
Also stop generating staging cluster configs, as well as prow core config and plugins

/assign chizhg albertomilan

### CODE REVIEW

File: guides/prow_setup.md [hunk 1]
```diff
@@ -43,14 +15,14 @@ resource manager service we use to manage a pool of GCP projects and handle the
...
+   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to create 10
    extra projects run
-   `./config/prod/prow/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`. You will
+   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`. You will
    need to substitute the actual billing ID for the second argument. In the
    event the script fails, it should be easy to follow along with in the GUI or
    run on the CLI. Projects are created with a numeric, incremental prefix
    automatically, based on the contents of
-   [prow/boskos_resources.yaml](../config/prod/prow/boskos/boskos_resources.yaml),
+   [build-cluster/boskos_resources.yaml](../config/prod/build-cluster/boskos/boskos_resources.yaml),
```
Comment (mattmoor) [inline guides/prow_setup.md:25]: Remove trailing whitespace:
```suggestion
   [build-cluster/boskos_resources.yaml](../config/prod/build-cluster/boskos/boskos_resources.yaml),
```

File: guides/prow_setup.md [hunk 2]
```diff
@@ -1,38 +1,10 @@
...
-1. After the PR is approved and merged, a staging process will be started to
-   make sure the changes do not break Prow. If it passes, the bot will create a
-   new PR to roll out the changes to
-   [config/prod/prow/cluster](../config/prod/prow/cluster). The new PR will be
-   automatically merged without needing manual approval.
-
-> Note all the config changes will be automatically updated on the Prow cluster
-> by a postsubmit Prow job, once the PR is merged.
+Prow are under [oss-test-infra
+repo](https://github.com/GoogleCloudPlatform/oss-test-infra/tree/master/prow/knative),
```
Comment (mattmoor) [inline guides/prow_setup.md:6]: Format markdown:
```suggestion
Prow are under
[oss-test-infra repo](https://github.com/GoogleCloudPlatform/oss-test-infra/tree/master/prow/knative),
```

File: guides/prow_setup.md [hunk 3]
```diff
@@ -43,14 +15,14 @@ resource manager service we use to manage a pool of GCP projects and handle the
...
 
 1. All projects and permissions can be created by running
-   `./config/prod/prow/create_boskos_projects.sh`. For example, to create 10
+   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to create 10
    extra projects run
-   `./config/prod/prow/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`. You will
+   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`. You will
    need to substitute the actual billing ID for the second argument. In the
    event the script fails, it should be easy to follow along with in the GUI or
    run on the CLI. Projects are created with a numeric, incremental prefix
```
Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

File: guides/prow_setup.md (hunk 1 above)
Comment (mattmoor) [inline guides/prow_setup.md:25]: Format markdown:
```suggestion
   [build-cluster/boskos_resources.yaml](../config/prod/build-cluster/boskos/boskos_resources.yaml),
```

File: guides/prow_setup.md (hunk 3 above)
Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

**Comment by chaodaiG** [comment]
/hold

Some of the custom jobs didn't run. @chizhg 
@peterfeifanchen for awareness

**Comment by chaodaiG** [comment]
> /hold
> 
> Some of the custom jobs didn't run. @chizhg
> @peterfeifanchen for awareness

nvm, false alarm
/unhold

File: guides/prow_setup.md (hunk 3 above)
Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

**Comment by chaodaiG** [comment]
@googlebot I consent.

**Comment by chaodaiG** [comment]
@googlebot I consent

File: guides/prow_setup.md (hunk 3 above)
Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

File: guides/manual_release.md [hunk 4]
```diff
@@ -145,66 +145,6 @@ Example:
 
 # Creating a new Knative release using Prow
 
-**Note:** The user starting the release job must have privileged access to the
```
Comment (chizhg) [inline guides/manual_release.md:148]: Nice! I saw this one day and also think we should delete it.

File: guides/prow_setup.md [hunk 5]
```diff
@@ -1,38 +1,10 @@
...
-All Prow related config files for [prow.knative.dev](https://prow.knative.dev)
-are under [config/prod/prow](../config/prod/prow). There is also a staging Prow
-in [prow-staging.knative.dev](https://prow-staging.knative.dev) for testing the
-new config changes before rolling out to production, and all its config files
-are under [config/staging/prow](../config/staging/prow).
-
-## Modify Prow cluster configs
-
 Prow is a collection of microservices that are deployed on a Kubernetes cluster.
 All the Kubernetes object `.yaml` files (except secrets) for Knative production
```
Comment (chizhg) [inline guides/prow_setup.md:4]: We shouldn't need the word `production` here as now we only have one Knative Prow.

File: tools/config-generator/main.go [hunk 6]
```diff
@@ -53,9 +51,6 @@ const (
 
 	// generalProwConfig contains config-wide definitions.
 	generalProwConfig = "prow_config.yaml"
```
Comment (chizhg) [inline tools/config-generator/main.go:53]: Can we also delete this line?

File: tools/config-generator/main.go [hunk 7]
```diff
@@ -694,23 +689,13 @@ func setOutput(fileName string) {
...
 func main() {
 	// Parse flags and sanity check them.
-	prowConfigOutput := ""
 	prowJobsConfigOutput := ""
 	testgridConfigOutput := ""
-	var env = flag.String("env", "prow", "The name of the environment, can be prow or prow-staging")
-	var generateProwConfig = flag.Bool("generate-prow-config", true, "Whether to generate the prow configuration file from the template")
-	var generatePluginsConfig = flag.Bool("generate-plugins-config", true, "Whether to generate the plugins configuration file from the template")
 	var generateTestgridConfig = flag.Bool("generate-testgrid-config", true, "Whether to generate the testgrid config from the template file")
 	var generateMaintenanceJobs = flag.Bool("generate-maintenance-jobs", true, "Whether to generate the maintenance periodic jobs (e.g. backup)")
```
Comment (chizhg) [inline tools/config-generator/main.go:695]: It looks a lot of flags here are not needed anymore, but it's fine by me to leave them for now.

File: tools/config-generator/templates/prow/prow_config.yaml [hunk 8]
```diff
@@ -1,152 +0,0 @@
-plank:
```
Comment (chizhg) [inline tools/config-generator/templates/prow/prow_config.yaml:1]: Did you delete `tools/config-generator/templates/prow/prow_plugins.yaml` as well? I don't see it in the changed files.

Comment (chizhg) [reply tools/config-generator/templates/prow/prow_config.yaml:1]: And also the generated config files for Prow and Prow staging?

File: guides/manual_release.md (hunk 4 above)
Comment (chaodaiG) [reply guides/manual_release.md:148]: :)

File: guides/prow_setup.md (hunk 5 above)
Comment (chaodaiG) [reply guides/prow_setup.md]: Done

File: tools/config-generator/main.go (hunk 6 above)
Comment (chaodaiG) [reply tools/config-generator/main.go:53]: Interesting, this file is gone on my branch

File: tools/config-generator/main.go (hunk 7 above)
Comment (chaodaiG) [reply tools/config-generator/main.go:695]: ack

File: guides/prow_setup.md [hunk 9]
```diff
@@ -1,38 +1,9 @@
...
-
-1. After the PR is approved and merged, a staging process will be started to
-   make sure the changes do not break Prow. If it passes, the bot will create a
-   new PR to roll out the changes to
-   [config/prod/prow/cluster](../config/prod/prow/cluster). The new PR will be
-   automatically merged without needing manual approval.
-
-> Note all the config changes will be automatically updated on the Prow cluster
-> by a postsubmit Prow job, once the PR is merged.
+All the Kubernetes object `.yaml` files (except secrets) for Knative prow are under
```
Comment (mattmoor) [inline guides/prow_setup.md:4]: Format markdown:
```suggestion
All the Kubernetes object `.yaml` files (except secrets) for Knative prow are
under
```

File: guides/prow_setup.md [hunk 10]
```diff
@@ -43,14 +14,14 @@ resource manager service we use to manage a pool of GCP projects and handle the
...
 
 1. All projects and permissions can be created by running
-   `./config/prod/prow/create_boskos_projects.sh`. For example, to create 10
+   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to create 10
    extra projects run
-   `./config/prod/prow/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`. You will
+   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`. You will
    need to substitute the actual billing ID for the second argument. In the
    event the script fails, it should be easy to follow along with in the GUI or
    run on the CLI. Projects are created with a numeric, incremental prefix
```
Comment (mattmoor) [inline guides/prow_setup.md:22]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

File: tools/config-generator/templates/prow/prow_config.yaml (hunk 8 above)
Comment (chaodaiG) [reply tools/config-generator/templates/prow/prow_config.yaml:1]: deleted in latest commit

File: guides/prow_setup.md (hunk 3 above)
Comment (mattmoor) [inline guides/prow_setup.md:23]: Format markdown:
```suggestion
   `./config/prod/build-cluster/create_boskos_projects.sh`. For example, to
   create 10 extra projects run
   `./config/prod/build-cluster/create_boskos_projects 10 0X0X0X-0X0X0X-0X0X0X`.
   You will need to substitute the actual billing ID for the second argument. In
   the event the script fails, it should be easy to follow along with in the GUI
   or run on the CLI. Projects are created with a numeric, incremental prefix
```

**Pull request merged**